.github/workflows/publish.yaml

```diff
@@ -0,0 +1,106 @@
+    strategy:
+      matrix:
+        containers:
+          - app: tftpd-hpa
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+
+      - name: Prepare
+        id: prepare
+        shell: bash
+        run: |
+          # Grab the version from the VERSION file
+          VERSION=$(cat ./apps/${{ matrix.containers.app }}/VERSION )
+
+          echo "OUTPUT_PROD_VERSION=$VERSION" >> $GITHUB_OUTPUT
+
+          # Initialize the output
+          OUTPUT_VERSION="$VERSION"
+
+          # If this is a pull request, append the PR number
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            OUTPUT_VERSION="$VERSION-pr${{ github.event.number }}"
+          fi
+
+          # Get the token
+          TOKEN=$(curl https://ghcr.io/token\?scope\="repository:truenas/${{ matrix.containers.app }}:pull" | jq '.token' --raw-output)
+
+          # Check if the "production" version already exists
+          EXISTS=$(curl -H "Authorization: Bearer $TOKEN" https://ghcr.io/v2/truenas/${{ matrix.containers.app }}/tags/list \
+          | jq --arg version "$VERSION" '.tags | index($version) != null')
+
+          # TODO: Check if Production version exists on DockerHub
+
+          # If "production" version exists...
+          if [[ $EXISTS == "true" ]]; then
+            # Set this output so we can trigger "failed" status
+            # on another step, to remind us to bump the version
+            echo "OUTPUT_EXISTS=true" >> $GITHUB_OUTPUT
+          fi
+
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            # Pretend the version doesn't exist, so we can build the PR tag
+            EXISTS="false"
+          fi
+
+          # If the version already exists, skip the build (When not a PR)
+          if [[ $EXISTS == "true" ]]; then
+            echo "Version $OUTPUT_VERSION already exists, skipping build"
+            exit 1
+          fi
+
+          # Set the output
+          echo "APP_VERSION=$OUTPUT_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Login to Github Container registry
+        if: steps.prepare.outputs.APP_VERSION != ''
+        uses: docker/login-action@40891eba8c2bcd1309b07ba8b11232f313e86779
+        with:
+          registry: ghcr.io
+          username: truenas
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Build and push Docker images
+        if: steps.prepare.outputs.APP_VERSION != ''
+        uses: docker/build-push-action@91df6b874e498451163feb47610c87c4a218c1ee
+        with:
+          context: apps/${{ matrix.containers.app }}/
+          push: true
+          # We can push to multiple registries
+          tags: |
+            ghcr.io/truenas/${{ matrix.containers.app }}:${{ steps.prepare.outputs.APP_VERSION }}
```

Comment:
We don't want to push to ghcr in this case and have people end up consuming either one and then one day we decide it's becoming a pain to maintain 2 different registeries and then we have to deal with these users.
Just use dockerhub please